maturin/import_hook/_file_lock.py

```diff
@@ -0,0 +1,192 @@
+import contextlib
+import errno
+import os
+import platform
+import time
+from abc import ABC, abstractmethod
+from pathlib import Path
+from types import ModuleType, TracebackType
+from typing import Optional, Type
+
+from maturin.import_hook._logging import logger
+
+fcntl: Optional[ModuleType] = None
+with contextlib.suppress(ImportError):
+    import fcntl
+
+
+msvcrt: Optional[ModuleType] = None
+with contextlib.suppress(ImportError):
+    import msvcrt
+
+
+class LockError(Exception):
+    pass
+
+
+class FileLock(ABC):
```

Comment:
Can't we use [filelock](https://pypi.org/project/filelock/) package instead?

Reply:
That's definitely an option. I didn't know if introducing dependencies would be 'allowed' since the original import hook did not have any

Reply:
Well we can add it as an optional dependency and document that import hook requires the extra dependency.

https://github.com/PyO3/maturin/blob/1b4db46085a5c171edfed452fdcc2478521b6275/pyproject.toml#L24-L30

for example, add:

```toml
import-hook = [
   "filelock",
]
```

Reply:
I switched to using filelock. It was fairly simple to do so